### PR TITLE
compiler-rt: Use self.sed_bindir to set LLVM_CONFIG_PATH

### PR DESCRIPTION
**Pull request by heshamelmatary**
Otherwise it fails to find it in the build dir with this error:

llvm-config failed with status No such file or directory

### CODE REVIEW

File: pycheribuild/projects/cross/compiler-rt.py [hunk 1]
```diff
@@ -47,14 +47,14 @@ class BuildCompilerRt(CrossCompileCMakeProject):
...
+        if self.target_info.is_rtems() or self.target_info.is_baremetal():
             self.add_cmake_options(CMAKE_TRY_COMPILE_TARGET_TYPE="STATIC_LIBRARY") # RTEMS only needs static libs
             # Get default target (arch) from the triple
             self.add_cmake_options(COMPILER_RT_DEFAULT_TARGET_ARCH=self.target_info.target_triple.split('-')[0])
 
         self.add_cmake_options(
-            LLVM_CONFIG_PATH=BuildCheriLLVM.getBuildDir(self, cross_target=CompilationTargets.NATIVE) / "bin/llvm-config",
-            LLVM_EXTERNAL_LIT=BuildCheriLLVM.getBuildDir(self, cross_target=CompilationTargets.NATIVE) / "bin/llvm-lit",
+            LLVM_CONFIG_PATH=self.sdk_bindir / "llvm-config",
+            LLVM_EXTERNAL_LIT=self.sdk_bindir/ "llvm-lit",
```
Comment (arichardson) [inline pycheribuild/projects/cross/compiler-rt.py:57]: This is probably better, but the libc++ jenkins job will need adjusting.

Comment (heshamelmatary) [reply pycheribuild/projects/cross/compiler-rt.py:57]: Is that why the CI check if failing?

Comment (arichardson) [reply pycheribuild/projects/cross/compiler-rt.py:57]: The failing CI reminds me why I made this change. For the native build there is no sdk_bindir, so it should use it from the build dir. Maybe the best solution is to change it to "build dir by default", but "sdk dir" for Jenkins.

File: pycheribuild/projects/cross/compiler-rt.py [hunk 2]
```diff
@@ -47,14 +47,14 @@ class BuildCompilerRt(CrossCompileCMakeProject):
...
-        if self.target_info.is_rtems():
+        if self.target_info.is_rtems() or self.target_info.is_baremetal():
             self.add_cmake_options(CMAKE_TRY_COMPILE_TARGET_TYPE="STATIC_LIBRARY") # RTEMS only needs static libs
             # Get default target (arch) from the triple
             self.add_cmake_options(COMPILER_RT_DEFAULT_TARGET_ARCH=self.target_info.target_triple.split('-')[0])
 
         self.add_cmake_options(
-            LLVM_CONFIG_PATH=BuildCheriLLVM.getBuildDir(self, cross_target=CompilationTargets.NATIVE) / "bin/llvm-config",
-            LLVM_EXTERNAL_LIT=BuildCheriLLVM.getBuildDir(self, cross_target=CompilationTargets.NATIVE) / "bin/llvm-lit",
+            LLVM_CONFIG_PATH=self.sdk_bindir / "llvm-config",
```
Comment (arichardson) [inline pycheribuild/projects/cross/compiler-rt.py:56]:
```suggestion
            LLVM_CONFIG_PATH=self.sdk_bindir / "llvm-config" if is_jenkins_build() else BuildCheriLLVM.getBuildDir(self, cross_target=CompilationTargets.NATIVE) / "bin/llvm-config",
```

**Comment by arichardson** [comment]
I guess the condition need to be `if is_jenkins_build() and not self.compiling_for_host()`

**Pull request merged**